python/cudf/cudf/_libxx/table.pyx

```diff
@@ -273,3 +252,24 @@ cdef mutable_table_view _make_mutable_table_view(columns) except*:
         mutable_column_views.push_back(col.mutable_view())
 
     return mutable_table_view(mutable_column_views)
+
+cdef columns_from_ptr(
+        unique_ptr[table] c_tbl,
+    ):
+        """
+        Return a list of table columns from a unique pointer
+
+        Parameters
+        ----------
+        c_tbl : unique_ptr[cudf::table]
+        """
+        num_columns = c_tbl.get().num_columns()
+        cdef vector[unique_ptr[column]] columns
+        columns = c_tbl.get()[0].release()
+        cdef vector[unique_ptr[column]].iterator it = columns.begin()
+
+        result = []
```

Comment:
lets empty initialize this so that Python doesn't have to go resizing the list constantly:
```suggestion
        result = [None] * num_cols
```

Reply:
Updated